tests/rust_usage_test/bin/alloc_check.rs

```diff
@@ -106,10 +106,10 @@ fn main() {
             assert_eq!("MyMonster", m.name());
 
             let pos = m.pos().unwrap();
-            assert_eq!(pos.x(), 1.0f32);
-            assert_eq!(pos.y(), 2.0f32);
-            assert_eq!(pos.z(), 3.0f32);
-            assert_eq!(pos.test1(), 3.0f64);
+            assert!((pos.x() - 1.0f32).abs() < std::f32::EPSILON);
```

Comment:
@CasperN Is using epsilon-tolerance equality the best choice here? I understand that, generally, we should compare floats with epsilon-tolerance. But, in this case, we know exactly what the bits are behind the values, so it seems to me that exact equality is preferred. cc @aardappel (maybe I'm overthinking this.)

Reply:
It is possible that the compiler optimizes `-` and `abs` in ways that produce subtle variations in float values, so yes, an epsilon is probably better. If anything, I'd wrap that in a dedicated `assert_close_eq` or something... though if theres few such cases, the current code is fine with me too.

Reply:
I agree that using epsilon isn't needed here, I'm just fixing the clippy lint and erring on the side of best practice. I thought about importing [`assert_approx_eq`](https://crates.io/crates/assert_approx_eq)  but yea, there weren't that many cases.

Reply:
@CasperN Please add a comment noting, essentially, that "we know these bits are exactly equal, but compilers may optimize floating-point code in subtle ways, so we play it safe and use epsilon comparison here" I'll be happy.